pages/sponsorship.js

```diff
@@ -0,0 +1,238 @@
+        this.setState({ sponsors: this.state.sponsors });
+        
+        // 3. refreshes form 
+        this.setState({
+            newobj: {"name":"", 
+                    "link": "",
+                    "umgURL": "",
+                    "lastmod": "",
+                    "tier": "",
+            }   
+        });   
+    }
+    
+    render() {
+        return(
+            <React.Fragment>
+                <Card>
+                    <CardHeader style={{backgroundColor: "#EDEDED"}}> 
+                        <CardTitle> Sponsors: {this.props.name} </CardTitle>
+                        <CardButtonContainer>
+                            <Button type={NEW} onClick={this.handleNew}>New Sponsor</Button>
+                        </CardButtonContainer>
+                    </CardHeader>
+
+                    <CardContent style={{backgroundColor: "F8F8F8"}}>
+                    <TableDiv>
+                        <CardDiv>
+                            <Text>
+                                Sponsor Name 
+                            </Text>
+                            <Text>
+                                Link
+                            </Text>
+                            <Text>
+                                Logo Image File
+                            </Text>
+                            <Text>
+                                Last Modified
+                            </Text>
+                            <Text>
+                                Actions
+                            </Text>
+                        </CardDiv>
+                          
+                        <div>
+                            {Object.values(this.state.sponsors).map((item)=> 
+                                <CardDiv key={item.name}>
+                                    <Text>
+                                        {item.name}
+                                    </Text>
+                                    <Text>  
+                                        {item.link}
+                                    </Text>
+                                    <Text>
+                                        {item.imgURL}
+                                    </Text>
+                                    <Text>
+                                        {item.lastmod}
+                                    </Text>
+                                    <Actions>
+                                        <button onClick={() => this.handleDelete(item.name)}> delete </button>
+                                        <button> view </button>
+                                        <button onClick={() => this.handleEdit(item.name)}> edit </button>
+                                    </Actions>
+                                </CardDiv>
+                            )}
+                        </div>
+                    </TableDiv>
+                    </CardContent>
+                </Card> 
+        
+                <form onSubmit={this.handleSave}>
```

Comment:
is the hide/show for this form finnished yet?

Reply:
not yet! almost done